pkg/storage/legacysql/dualwrite/kvstore.go

```diff
@@ -0,0 +1,83 @@
+package dualwrite
+
+import (
+	"context"
+	"encoding/json"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/grafana/grafana-app-sdk/logging"
+	"github.com/grafana/grafana/pkg/infra/kvstore"
+)
+
+type keyvalueDB struct {
+	db     kvstore.KVStore
+	logger logging.Logger
+}
+
+const globalKVOrgID = 0
+const globalKVNamespace = "unified.dualwrite" // Does not conflict with "storage.dualwriting" used for requested state
+
+func newKeyValueDB(kv kvstore.KVStore) *keyvalueDB {
+	return &keyvalueDB{
+		db:     kv,
+		logger: logging.DefaultLogger.With("logger", "dualwrite.db"),
+	}
+}
+
+func (m *keyvalueDB) Get(ctx context.Context, gr schema.GroupResource) (status StorageStatus, ok bool, err error) {
```

Comment:
let's add some tests :)

Reply:
I changed this to internal -- since the get/set methods are not externally accessible and just abstracts the storage for the service (that already tests the get/set behavior)